projects/components/src/table/table.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable, Injector } from '@angular/core';
+import { FilterAttribute } from '../filter-bar/filter-attribute';
+import { FilterType } from '../filter-bar/filter-type';
+import { TableCellParserLookupService } from './cells/table-cell-parser-lookup.service';
+import { TableCellRendererLookupService } from './cells/table-cell-renderer-lookup.service';
+import { CoreTableCellRendererType } from './cells/types/core-table-cell-renderer-type';
+import { TableCdkDataSource } from './data/table-cdk-data-source';
+import { TableColumnConfig, TableColumnConfigExtended } from './table-api';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TableService {
+  private static readonly STATE_ATTRIBUTE: FilterAttribute = {
+    name: '$$state',
+    displayName: 'Row State',
+    type: '$$state' as FilterType
+  };
+
+  public constructor(
+    private readonly injector: Injector,
+    private readonly tableCellRendererLookupService: TableCellRendererLookupService,
+    private readonly tableCellParserLookupService: TableCellParserLookupService
+  ) {}
+
+  public buildExtendedColumnConfigs(
+    columnConfigs: TableColumnConfig[],
+    dataSource: TableCdkDataSource,
+    attributes: FilterAttribute[] = []
+  ): TableColumnConfigExtended[] {
+    return columnConfigs.map(columnConfig => {
+      const attribute = this.isStateColumnConfig(columnConfig)
+        ? TableService.STATE_ATTRIBUTE
+        : attributes.find(attr => attr.name === columnConfig.name);
+
+      const rendererConstructor = this.tableCellRendererLookupService.lookup(
+        columnConfig.display !== undefined ? columnConfig.display : CoreTableCellRendererType.Text
+      );
+
+      const parserConstructor = this.tableCellParserLookupService.lookup(rendererConstructor.parser);
+
+      return {
+        ...columnConfig,
+        attribute: attribute,
+        renderer: rendererConstructor,
+        parser: new parserConstructor(this.injector),
```

Comment:
It feels cleaner to make the parsers just services and inject instances of them, rather than plain classes that we're tossing instances of an injector. They're stateless right?

Reply:
Eh, I'm not so sure I agree on this one. But when I move parsers out of table I will explore the idea of making them injectable at that time.

Reply:
In terms of making them injectable, wouldn't it just be marking them as `@Injectable` ? Then, we don't need to new it, don't need the base constructor param since each parser is independent, etc.
Then here, instead of newing, `injector.get(parserConstructor)`. But I guess that's predicated on thinking that's the right path.

Reply:
I'll definitely explore. Not happy with the `new` either, but it seems the most light weight. Maybe trying to switch things over will convince me its not too bad.

Reply:
`new ` and `.get` is the same number of chars. just sayin'